### ISSUE TITLE
Question about stable accuracy on CIFAR100

### ISSUE THREAD

**Issue by HowieMa** [huawei-noah/Data-Efficient-Model-Compression#16]
Thank you for providing the code for your amazing work! I have some question about the accuracy of experiments on CIFAR 100. 

Firstly, I can get similar results of CIFAR-10 with scripts on the [README.md](https://github.com/huawei-noah/Data-Efficient-Model-Compression/blob/master/DAFL/README.md). Like my accuracy of teacher is 0.9518, and that of DAFL is 0.9241. 

However, I cannot get the accuracy around 74% of DAFL in CIFAR-100 as in your paper. I follow the same parameters in readme. That is, 
`python DAFL-train.py --dataset cifar100 --channels 3 --n_epochs 2000 --batch_size 1024 --lr_G 0.02 --lr_S 0.1 --latent_dim 1000 --oh 0.05 --ie 20 --act 0.01`. 

I run the DAFL code twice, the first time I got  53.31% accuracy, while the second time I got 62.39%. Both of them are far away from 74% in the Table. I did use your teacher training script and the accuracy is 0.771, which is a little bit lower than yours. And I did use 1024 batch size here, as I know that a small batch size may cause a huge difference. 
 
The log for my second training is
~~~
[Epoch 1/2000] [loss_oh: 1.565716] [loss_ie: -1.733893] [loss_a: -0.346861] [loss_kd: 0.835829]
[Epoch 2/2000] [loss_oh: 1.836624] [loss_ie: -1.873039] [loss_a: -0.330029] [loss_kd: 0.749510]
...
[Epoch 799/2000] [loss_oh: 1.751102] [loss_ie: -1.998407] [loss_a: -0.336999] [loss_kd: 0.165117]
[Epoch 800/2000] [loss_oh: 1.767649] [loss_ie: -1.998450] [loss_a: -0.337684] [loss_kd: 0.163865]
[Epoch 801/2000] [loss_oh: 1.750000] [loss_ie: -1.998442] [loss_a: -0.337093] [loss_kd: 0.126873]
...
[Epoch 1598/2000] [loss_oh: 1.734455] [loss_ie: -1.997838] [loss_a: -0.337273] [loss_kd: 0.084451]
[Epoch 1599/2000] [loss_oh: 1.737263] [loss_ie: -1.998453] [loss_a: -0.337822] [loss_kd: 0.080239]
[Epoch 1600/2000] [loss_oh: 1.697831] [loss_ie: -1.998273] [loss_a: -0.337703] [loss_kd: 0.081537]
...
[Epoch 1997/2000] [loss_oh: 1.739230] [loss_ie: -1.998205] [loss_a: -0.337063] [loss_kd: 0.053484]
[Epoch 1998/2000] [loss_oh: 1.729897] [loss_ie: -1.998376] [loss_a: -0.337201] [loss_kd: 0.051445]
[Epoch 1999/2000] [loss_oh: 1.730324] [loss_ie: -1.997864] [loss_a: -0.337307] [loss_kd: 0.053925]
~~~
Also, this model achieved 0.4108 around 800 epochs, and 0.5790 around 1600 epochs. 

I run it with Pytorch 1.4 + cuda 10.1. 

It seems that the results are not very stable on CIFAR-100. Thus, I would like to know whether this thing is normal on CIFAR-100. And I also want to know whether my 62% accuracy is reasonable. 

By the way, could you please share your training log of CIFAR-100 and hyper parameters? 

I really appreciate it if you could help me! Look forward to your reply.

**Comment by HantingChen** [comment]
Sorry for the late reply.

The hyper-parameter is wrong during some updates in version. 

So the correct parameters are:

python DAFL-train.py --dataset cifar100 --channels 3 --n_epochs 2000 --batch_size 1024 --lr_G 0.02 --lr_S 0.1 --latent_dim 1000 --oh 0.5 --ie 20 --act 0.1

**Comment by HowieMa** [comment]
> Sorry for the late reply.
> 
> The hyper-parameter is wrong during some updates in version.
> 
> So the correct parameters are:
> 
> python DAFL-train.py --dataset cifar100 --channels 3 --n_epochs 2000 --batch_size 1024 --lr_G 0.02 --lr_S 0.1 --latent_dim 1000 --oh 0.5 --ie 20 --act 0.1

Thanks for answering my question!

**Comment by planemanner** [comment]
@HowieMa Did you get the result of paper when you tune the hyper-parameters as you've been recommended?

**Comment by HowieMa** [comment]
> @HowieMa Did you get the result of paper when you tune the hyper-parameters as you've been recommended?
In my experiment, my teacher network ResNet34 achieves 0.771, and the DAFL achieves 0.7106. Although there is a small gap between the result in paper and my result, I think it is reasonable to some degree.
